src/wallet3/default_daemon_comms.cpp

```diff
@@ -30,4 +197,21 @@ namespace wallet
     wallets.push_back(wallet.shared_from_this());
   }
 
+  void
+  DefaultDaemonComms::ForEachWallet(std::function<void(std::shared_ptr<Wallet>)> func)
```

Comment:
i feel like this should be a templated function in the header?

Reply:
If feasible then this definitely should happen because `std::function` are inherently unoptimizable because of the type erasure they do, whereas a templated call function is (so that the compiler can inline a tiny lambda, etc.).

```c++
template <typename F>
void ForEachWallet(F&& callable) { /*...*/ callable(whatever); /* ... */ }
```